decidim-core/lib/decidim/form_builder.rb

```diff
@@ -251,20 +251,18 @@ def check_box(attribute, options = {}, checked_value = "1", unchecked_value = "0
     def date_field(attribute, options = {})
       value = object.send(attribute)
       data = { datepicker: "" }
-      data[:startdate] = I18n.localize(value, format: :datepicker) if value.present?
-      iso_value = value.present? ? value.strftime("%Y-%m-%d") : ""
+      data[:startdate] = I18n.localize(value, format: :datepicker) if value.present? && [Date, DateTime, Time].include?(value.class)
+      displayed_format = I18n.t("date.formats.displayed")
+      data[:"date-format"] = displayed_format
 
       template = ""
       template += label(attribute, label_for(attribute) + required_for_attribute(attribute))
       template += @template.text_field(
         @object_name,
         attribute,
-        options.merge(name: nil,
-                      id: "date_field_#{@object_name}_#{attribute}",
-                      data: data)
+        options.merge(data: data)
       )
-      template += @template.hidden_field(@object_name, attribute, value: iso_value)
-      template += error_and_help_text(attribute, options)
+      template += error_and_help_text(attribute, options.merge(help_text: displayed_format))
```

Comment:
I would enhance the help text a bit, to say "Expected format: mm/dd/yyyy", or something like that, not just the raw expected format.

By the way, I tried the placeholder approach and it looked great on my browser. But the fact that it wasn't very visible on yours makes it clear that help text is better :+1:.

Reply:
Also, you should modify the `datetime_field` method below in the same way you modified this one, so that manual edition works there too, and we don't use different approaches, which will be confusing.

Reply:
Although I find it self explanatory I'll create a text for the help text , not only the format, 

Regarding the `datetime_field`, it would be better to solve it in another PR.

Reply:
Yeah, the tweak in the help text is not very important. Feel free to not do it, I don't have a strong opinion.

Regarding `datetime_field`, I have to disagree here. I proposed the alternative solution of fixing the bug while keeping the hidden field. In my opinion, that was easier to implement since most of the work is already done. But you made your point that your approach is better and I agree. However, if we do it, we should migrate both methods, not just one. I does not make any sense to use a hidden field for timepicker fields, and a backend conversion for datepicker fields. Specially when you have justified your approach saying "I'd rather remove this field that causes confusion". If we leave it like that it will be even more confusing because it will only be used on time fields but not on date fields. I would fix it now.